### PR TITLE
Bug: PVS issues in two-player mode

### PR DESCRIPTION
**Pull request by 0xABADCAFE**
Fixes issues with edge PVS caused by reliance purely on the Player 1 positional data.

### CODE REVIEW

**Comment by mheyer32** [comment]
This pull request includes several changes to the `ab3d2_source` project, focusing on improvements in zone handling and visibility calculations. The most important changes include the addition of new data structures and functions for determining zone crossings, updates to visibility handling, and adjustments to various assembly files to accommodate these changes.

### Improvements to zone handling and visibility calculations:

* [`ab3d2_source/c/zone.h`](diffhunk://#diff-9adc86c1e8a4426d69b5d525808cf701cf5d6c799d972b527e97d8e4ca6da681L194-R223): Added `ZoneCrossing` enumeration and the `Zone_DetermineCrossing` function to determine possible crossings between zones based on their heights.
* [`ab3d2_source/c/zone_edge_pvs.c`](diffhunk://#diff-641823decf70e89b6f533da8ed680b7217dae46caa7f46c366977698df743feaL100-R110): Updated functions to use `Zone_DetermineCrossing` for calculating visibility and joining edges. Added `visJoinMask` to track visible joins. [[1]](diffhunk://#diff-641823decf70e89b6f533da8ed680b7217dae46caa7f46c366977698df743feaL100-R110) [[2]](diffhunk://#diff-641823decf70e89b6f533da8ed680b7217dae46caa7f46c366977698df743feaR213-R224) [[3]](diffhunk://#diff-641823decf70e89b6f533da8ed680b7217dae46caa7f46c366977698df743feaR598-R599) [[4]](diffhunk://#diff-641823decf70e89b6f533da8ed680b7217dae46caa7f46c366977698df743feaR630) [[5]](diffhunk://#diff-641823decf70e89b6f533da8ed680b7217dae46caa7f46c366977698df743feaR660) [[6]](diffhunk://#diff-641823decf70e89b6f533da8ed680b7217dae46caa7f46c366977698df743feaR673) [[7]](diffhunk://#diff-641823decf70e89b6f533da8ed680b7217dae46caa7f46c366977698df743feaR684-R686)
* [`ab3d2_source/modules/draw/draw_zone_graph.s`](diffhunk://#diff-758534b4b1c58448d95078a621564562b0747abbfaa681a7e17112e788e688cbR1-R10): Added `draw_Root_Zone_w` and modified the zone graph drawing logic to handle upper and lower zones more accurately. [[1]](diffhunk://#diff-758534b4b1c58448d95078a621564562b0747abbfaa681a7e17112e788e688cbR1-R10) [[2]](diffhunk://#diff-758534b4b1c58448d95078a621564562b0747abbfaa681a7e17112e788e688cbL67-R74) [[3]](diffhunk://#diff-758534b4b1c58448d95078a621564562b0747abbfaa681a7e17112e788e688cbR186-R194) [[4]](diffhunk://#diff-758534b4b1c58448d95078a621564562b0747abbfaa681a7e17112e788e688cbR208-R211) [[5]](diffhunk://#diff-758534b4b1c58448d95078a621564562b0747abbfaa681a7e17112e788e688cbR246) [[6]](diffhunk://#diff-758534b4b1c58448d95078a621564562b0747abbfaa681a7e17112e788e688cbR312)

### Codebase adjustments:

* [`ab3d2_source/hires.s`](diffhunk://#diff-f7558a3346c4096a81e7a870dfebaed8b346e43a32d99ee020c7cf69bd71f722L1179-R1179): Renamed `SplitHeight` to `Zone_SplitHeight_l` to better reflect its purpose and updated relevant sections of the code. [[1]](diffhunk://#diff-f7558a3346c4096a81e7a870dfebaed8b346e43a32d99ee020c7cf69bd71f722L1179-R1179) [[2]](diffhunk://#diff-f7558a3346c4096a81e7a870dfebaed8b346e43a32d99ee020c7cf69bd71f722L1312-R1312) [[3]](diffhunk://#diff-f7558a3346c4096a81e7a870dfebaed8b346e43a32d99ee020c7cf69bd71f722L1428-R1428) [[4]](diffhunk://#diff-f7558a3346c4096a81e7a870dfebaed8b346e43a32d99ee020c7cf69bd71f722L3354-R3354)
* [`ab3d2_source/kmake.sh`](diffhunk://#diff-97b885fe492915e32ca37e23ddf282773bc647e2fc341c21e0eb05babc2884d8L4-R12): Modified the build script to differentiate between release and development builds with appropriate naming conventions.

These changes enhance the handling of zones and visibility within the project, making the codebase more robust and maintainable.

**Comment by mheyer32** [comment]
Oups, let me explain. I was trying out the Copilot function to summarize the change (and the above text is what came out of it). But then I accidentally closed the PR because my brain read "Close comment" instead of "Close _with_ comment".

**Comment by 0xABADCAFE** [comment]
Unfortunately I was fixing forwards that PVS bug for 2 player mode - the other changes are incremental improvements towards the next step of PVS work which will be the door/lift problem. However, the zone crossing work is still relevant for some scenarios regardless of the door/lift issue.

File: ab3d2_source/c/zone_edge_pvs.c [hunk 1]
```diff
@@ -513,8 +531,13 @@ void Zone_UpdateVectors() {
...
     //dputs("Zone_UpdateVectors()");
-    zone_ViewPoint.v_X   = Plr1_Position_vl[POS_X];
-    zone_ViewPoint.v_Z   = Plr1_Position_vl[POS_Z];
+    if (Plr_MultiplayerType_b == MULTIPLAYER_SLAVE) {
+        zone_ViewPoint.v_X   = Plr2_Position_vl[POS_X];
+        zone_ViewPoint.v_Z   = Plr2_Position_vl[POS_Z];
+    } else {
+        zone_ViewPoint.v_X   = Plr1_Position_vl[POS_X];
+        zone_ViewPoint.v_Z   = Plr1_Position_vl[POS_Z];
+    }
```
Comment (0xABADCAFE) [inline ab3d2_source/c/zone_edge_pvs.c:540]: This is the _actual_ fix for the two player PVS issue.

**Comment by 0xABADCAFE** [comment]
It does seem like copilot made a reasonable attempt at summarising the changes

**Comment by 0xABADCAFE** [comment]
Myself and @AndyLoft have both been able to test that the 2 player mode seems to be working now, as is the extra non-crossable joining edge test, so I'm going to merge as is.

**Pull request merged**